### DISCUSSION TITLE
OpenSSL with ZLIB_SHARED

### DISCUSSION THREAD

**Discussion by gvanem** [cesanta/mongoose#1931] [Ask questions, get answers] [answered]
I'm not sure this is an issue with Mongoose or my build of OpenSSL.
But running the `unit_test.exe` on Windows, it crashes in `test_tls()` deep inside OpenSSL
calling `(*p_compress)(out, &out_size, in, ilen)`:
```
0x0
libcrypto_3!zlib_oneshot_compress_block+0x45
libcrypto_3!COMP_compress_block+0x3d
libssl_3!tls_construct_client_compressed_certificate+0x1a6
libssl_3!write_state_machine+0x1a8
libssl_3!state_machine+0x342
libssl_3!SSL_do_handshake+0x109
unit_test!mg_tls_handshake+0x36
unit_test!connect_conn+0x24d
unit_test!mg_mgr_poll+0x54c
unit_test!fetch+0x13d
unit_test!test_tls+0xae
```

I see `p_comp` is NULL for some reason. How / who is responsible for checking that this BIO-method is ok?

**Comment by cpq** [comment, marked as answer]
@gvanem this belongs to issues, IMO.
Could you rebuild with symbols and re-send the backtrace with line numbers, params, etc, please?
